features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/ButtonCompositionGroupMapper.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.compose.internal.mappers
+
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.ButtonColors
+import androidx.compose.ui.geometry.Size
+import com.datadog.android.sessionreplay.compose.internal.data.Box
+import com.datadog.android.sessionreplay.compose.internal.data.ComposableParameter
+import com.datadog.android.sessionreplay.compose.internal.data.ComposeWireframe
+import com.datadog.android.sessionreplay.compose.internal.data.UiContext
+import com.datadog.android.sessionreplay.compose.internal.reflection.accessible
+import com.datadog.android.sessionreplay.model.MobileSegment
+
+internal class ButtonCompositionGroupMapper : AbstractCompositionGroupMapper() {
+
+    override fun map(
+        stableGroupId: Long,
+        parameters: Sequence<ComposableParameter>,
+        boxWithDensity: Box,
+        uiContext: UiContext
+    ): ComposeWireframe? {
+        var cornerRadius: Number? = null
+        var enabled: Boolean? = null // TODO use ?
+        var colors: ButtonColors? = null
+
+        parameters.forEach { param ->
+            when (param.name) {
+                "colors" -> colors = param.value as? ButtonColors
+                "enabled" -> enabled = param.value as? Boolean
+                "shape" -> cornerRadius = (param.value as? RoundedCornerShape)?.let {
+                    it.topStart.toPx(Size.Unspecified, uiContext.composeDensity) / uiContext.density
+                }
+            }
+        }
+
+        var backgroundColor: String? = null
+        var contentColor: String? = uiContext.parentContentColor
+
+        colors?.let { c ->
+            // TODO memoize fields
```

Comment:
If leaving for later, should we not add `RUM-0000` on this TODO/above on `enabled` prop TODO?